### ISSUE TITLE
Error while configuring engine: The recipe "Blog" could not be found

### ISSUE THREAD

**Issue by khaledhikmat** [statiqdev/Statiq#606]
Hi, I tried Wyam last year and got it working on a beta release. I wanted to try the release version on something new so I downloaded v1.1.0  Zip file, extract it and wanted to follow the instructions as in this blog: [https://win32.io/posts/Wyam-GitHub-Pages](https://win32.io/posts/Wyam-GitHub-Pages). 

Using the first command I tried: `wyam new -r Blog`, I am getting this error message:
**Error while configuring engine: The recipe "Blog" could not be found**

What am I doing wrong? Thanks.

**Comment by daveaglick** [comment]
You might be hitting #587 If an old version of the recipe package is sitting around in your NuGet cache, Wyam sometimes tries to use that instead of updated packages (#587 is to take another look at this behavior). Try running it with the -u flag. If that still doesn’t work, please run with -v and upload the resulting log here.

**Comment by khaledhikmat** [comment]
Yeah....thanks....that worked. However, I seem to be running into a bigger problem which is what initially promoted me to try the release version. Until last week, I was able to run `yam build` and try my site locally....I would do this every once in a while and it always worked. Yesterday, I added a new post and did the same thing ...but the build process hung and the CPU utilization went off the roof. 

Using the task manager, I could not end the task....I had to restart the machine. I tried it again....the same thing except that Norton said that wyam.exe is acting suspiciously and it has to be deleted. So I re-installed the beta version from an old zip I had and re-tried....the same thing.

Then I downloaded v1.1.0 and tried again. I got into the above problem. Then with your recommendation above, the build worked ...so I added my posts and tried again and things work normally again. The I changed the `config.wyam` file and re-build....it stuck, CPU off the roof and `Wyam.exe` cannot be terminated. Restarted.....removed the last post that I added, put back the `config.wyam` the way it was....tried it ...same problem. 

I am not really sure what is going on....but something is very wrong. It could be something in my files that is causing an infinite loop somewhere. Not sure if you are interested in solving this problem....obviously it is not happening with everyone...so it could be my data. I can send you my entire directory if you like. Let me know please. Thanks for your help.

**Comment by daveaglick** [comment]
Sure, I'd be happy to take a quick look. Is the site on a public repo? If not, feel free to zip and attach here or send it to me.

**Comment by khaledhikmat** [comment]
Ok...great...thank you sir. Here is the link:
[https://www.dropbox.com/s/rtj1wfj77fnohf3/wyam-blog.zip?dl=0](https://www.dropbox.com/s/rtj1wfj77fnohf3/wyam-blog.zip?dl=0)

Regards

**Comment by khaledhikmat** [comment]
Hi, back to this.....I tried again....this time I used a brand new machine. I am getting this error below. Thanks.

```
**BUILD**
Loading configuration from file:///D:/Sites/Wyam/khaledhikmat/config.wyam
Installing NuGet packages
    Installing packages to C:/Users/admin/.nuget/packages (using global packages folder)
        Cached package Wyam.Blog.CleanBlog 1.1.0 does not exist in packages folder
    NuGet packages installed in 216786 ms
Recursively loading assemblies
    Assemblies loaded in 3709 ms
Cataloging classes
    Classes cataloged in 304 ms
Error while loading configuration: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'Wyam.Blog.Blog' threw an exception. ---> System.MissingMethodException: Method not found: 'Wyam.Core.Modules.Control.Concat Wyam.Core.Modules.Control.Concat.Where(System.Func`4<Wyam.Common.Execution.IExecutionContext,Wyam.Common.Documents.IDocument,System.Collections.Generic.IReadOnlyList`1<Wyam.Common.Documents.IDocument>,Boolean>)'.
   at Wyam.Web.Pipelines.Pages.GetModules(PagesSettings settings)
   at Wyam.Web.Pipelines.Pages..ctor(String name, PagesSettings settings)
   at Wyam.Blog.Blog..cctor()
   --- End of inner exception stack trace ---
   at Wyam.Blog.Blog.get_Pages()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at Wyam.Common.Configuration.Recipe.<Apply>b__0_4(<>f__AnonymousType0`2 x) in E:\Code\Wyam\src\core\Wyam.Common\Configuration\Recipe.cs:line 54
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Wyam.Common.Configuration.Recipe.Apply(IEngine engine) in E:\Code\Wyam\src\core\Wyam.Common\Configuration\Recipe.cs:line 61
   at Wyam.Blog.Blog.Apply(IEngine engine)
   at Wyam.Configuration.Configurator.ApplyRecipe() in E:\Code\Wyam\src\core\Wyam.Configuration\Configurator.cs:line 314
   at Wyam.Configuration.Configurator.Configure(String script) in E:\Code\Wyam\src\core\Wyam.Configuration\Configurator.cs:line 141
   at Wyam.EngineManager.Configure() in E:\Code\Wyam\src\clients\Wyam\EngineManager.cs:line 125
```

**Comment by daveaglick** [comment]
Sorry I dropped the ball on this issue. Hopefully it's resolved with the 2.0 release - there's also been some work since this was opened on ensuring the correct packages that match the running Wyam version get picked up. I'm going to go ahead and close this issue - please feel free to request it be reopened or to open a new issue if you continue to have problems.
